crates/core/tedge/src/cli/certificate/upload.rs

```diff
@@ -44,8 +44,18 @@ impl UploadCertCmd {
             Err(_) => rpassword::read_password_from_tty(Some("Enter password: "))?,
         };
 
-        // Use a builder instead of `Client::new`, `new` could panic, builder adds option to allow invalid certs.
-        let client = reqwest::blocking::Client::builder().build()?;
+        let config = get_tedge_config()?;
+        let root_cert = config.query(C8yRootCertPathSetting)?;
+        let client_builder = reqwest::blocking::Client::builder();
+        let client = match std::fs::metadata(&root_cert)?.is_file() {
+            true => {
+                dbg!("its file");
+                let cert = std::fs::read(root_cert)?;
+                let cert_pem = reqwest::Certificate::from_pem(&cert)?;
+                client_builder.add_root_certificate(cert_pem).build()?
+            }
+            false => client_builder.build()?,
```

Comment:
If the `root_cert_path` is a directory, like `etc/ssl/certs`, the `client_builder` doesn't need that path to be set explicitly?

Reply:
No need because the `rustls` reads the `etc/ssl/certs` directory. The problem is it can not read the `self-signed` certificate. So, this has to be added explicitly.

Reply:
In that case I don't understand this fix. Are you saying that, if someone has installed the self-signed certificate into `/etc/ssl/certs` directory and invokes `tedge cert upload` to thick-edge, it won't work? So, anyone wanting to connect to thick edge must set the tedge `root.cert.path` to the certificate **file** of thick edge and not **can not use** `/etc/ssl/certs` dir?

Reply:
Yes, that's right. Even if its installed inside the `/etc/ssl/certs` the certificate is not read, validated and added into the `rustls root cert store`. So, need to set the path to this file explicitly using `c8y.root.cert.path` and added to the `rustls root store` explicitly.  And this has to be documented.

Reply:
Yes, it definitely needs to be documented. It need not be documented as a Thick Edge specific workaround, but rather applicable to any Cumulocity deployment (like private cloud deployments) that uses self-signed certificates.